Makefile

```diff
@@ -122,7 +123,7 @@ $(TESSERACT_PREFIX)/bin/tesseract: build/tesseract/Makefile
 $(TESSERACT_PREFIX)/bin/lstmtraining: build/tesseract/Makefile
 	$(MAKE) -C build/tesseract training-install
 
-TESSERACT_CONFIG ?= --disable-openmp --disable-shared CXXFLAGS="-g -O2 -fPIC -fno-math-errno -Wall -Wextra -Wpedantic"
+TESSERACT_CONFIG ?= --disable-openmp --disable-shared CXXFLAGS="-g -O2 -fPIC -fno-math-errno -Wall -Wextra -Wpedantic -UNDEBUG"
```

Comment:
@bertsky, it's a bit strange if you add an unrelated commit to a pull request which was already reviewed by yourself.

I am not sure that your change in line 126 is a good idea (debug code adds instructions which add processing time), and you did not explain why you made it.

Reply:
I'm afraid your change will do exactly what you wanted to avoid.

Reply:
see [this](https://github.com/tesseract-ocr/tesseract/blob/main/src/ccutil/errcode.cpp#L72-L83) why we must at least avoid `NDEBUG`; keeping `-g` otherwise is a good idea to get symbols when there is a crash

all this is a far cry from being satisfactory (libtesseract must use exceptions in the end!), just less insane (segfault)

Reply:
So you prefer getting a SIGSEGV because of your change instead of getting an abort() like in the current configuration?

Reply:
in other places like [this](https://github.com/tesseract-ocr/tesseract/blob/68caca6951b384ef5882e98b64198964a37af488/src/ccutil/elst.h#L563-L566) we need `!defined(NDEBUG)` for certain safety checks

Reply:
> So you prefer getting a SIGSEGV because of your change instead of getting an abort() like in the current configuration?

No, segfault is obviously even worse, but I'm in a bind: the place that actually triggers my latest segfault/abort needs `!defined(NDEBUG)` – so the snippet I showed first behaves different, but is (currently) not as relevant